### PR TITLE
Upgraded Cypress to 13.3.0

### PR DESCRIPTION
**Pull request by alfredeen**
## Description

This PR upgrades Cypress to version 13 and the cypress github action to v6. It attempts to correct some flaky tests concerning app deployments and the public apps page.

It introduces a new custom command **logf** to be used instead of the built-in cy.log for better troubleshooting Use it like this:

cy.logf("The log message", Cypress.currentTest)

Jira [link](https://scilifelab.atlassian.net/browse/SS-1095)

## Checklist

- [x] This pull request is against **develop** branch (not applicable for hotfixes)
- [x] I have included a link to the issue on GitHub or JIRA (if any)
- [ ] I have included migration files (if there are changes to the model classes)
- [x] I have included, reviewed and executed tests (unit and end2end) to complement my changes
- [ ] I have updated the related documentation (if necessary)
- [x] I have added a reviewer for this pull request
- [x] I have added myself as an author for this pull request
- [ ] In the case I have modified settings.py, then I have also updated the studio-settings-configmap.yaml file in serve-charts

### CODE REVIEW

**Review by churnikov — COMMENTED** [review]
Everything looks good to me, I just have some questions.
I see that e2e test fail though because of timeout. I'd approve PR if not that.

**Comment by alfredeen** [comment]
> Everything looks good to me, I just have some questions. I see that e2e test fail though because of timeout. I'd approve PR if not that.

That the e2e tests now fail is a good thing, because they should fail before the fix to app status is merged.

File: cypress/e2e/ui-tests/test-deploy-app.cy.js
Comment (alfredeen) [reply cypress/e2e/ui-tests/test-deploy-app.cy.js:11]: Yes but we had set a very long timeout for all commands, even just opening pages. I now have 2 timeouts, one normal for most commands and one very long 3 min timeout for k8s operations.

File: cypress/e2e/ui-tests/test-deploy-app.cy.js
Comment (alfredeen) [reply cypress/e2e/ui-tests/test-deploy-app.cy.js:15]: It doesn't but it includes the current test initiating the log message in the output and the clock time for tracking chronological order.

File: cypress/support/e2e.js
Comment (alfredeen) [reply cypress/support/e2e.js:22]: Yes, this was the fix :-)

File: cypress/e2e/ui-tests/test-superuser-functionality.cy.js
Comment (churnikov) [inline cypress/e2e/ui-tests/test-superuser-functionality.cy.js:177]: Why keeping it if not needed?:)

**Review by churnikov — APPROVED** [review]

**Pull request merged**
